lab/MusicalTree/MusicalTree.js

```diff
@@ -0,0 +1,275 @@
+import { Point } from "../../pga2d/objects.js";
+import { Color } from "../../sketchlib/Color.js";
+import { WIDTH, HEIGHT } from "../../sketchlib/dimensions.js";
+import { LSystem } from "../../sketchlib/LSystem.js";
+import { draw_primitive } from "../../sketchlib/p5_helpers/draw_primitive.js";
+import { GroupPrimitive } from "../../sketchlib/rendering/GroupPrimitive.js";
+import { LinePrimitive } from "../../sketchlib/rendering/primitives.js";
+import { group, style } from "../../sketchlib/rendering/shorthand.js";
+import { Style } from "../../sketchlib/Style.js";
+import { CanvasMouseHandler } from "../lablib/CanvasMouseHandler.js";
+import { MouseInput } from "../lablib/MouseInput.js";
+import { N16, N32 } from "../lablib/music/durations.js";
+import { B4, C4, E4, E6, F6, G6 } from "../lablib/music/pitches.js";
+import { Melody, Note, Rest, Score } from "../lablib/music/Score.js";
+import { MuteButton } from "../lablib/MuteButton.js";
+import { PlayButtonScene } from "../lablib/PlayButtonScene.js";
+import { Rational } from "../lablib/Rational.js";
+import { SoundManager } from "../lablib/SoundManager.js";
+
+const TREE_LSYSTEM = new LSystem("Fa", {
+  a: "[+Fa][-Fa]",
+});
+
+function count_symbols(str) {
+  /**
+   * @type {Map<string, number>}
+   */
+  const counts = new Map();
+  for (const c of str) {
+    const current_count = counts.get(c) ?? 0;
+    counts.set(c, current_count + 1);
+  }
+
+  return counts;
+}
+
+const TREE_STRS = TREE_LSYSTEM.iterate(8);
+for (const str of TREE_STRS) {
+  const counts = count_symbols(str);
+  const ignore_a_len = str.length - counts.get("a");
+  console.log(ignore_a_len, counts, str);
+}
+
+/**
+ *
+ * @param {string} tree_str
+ * @returns {GroupPrimitive}
+ */
+function render_tree(tree_str) {
+  const DELTA_ANGLE = Math.PI / 4;
+  const MAX_LENGTH = 100;
+  const LENGTH_SCALE = 0.7;
+  let position = Point.point(WIDTH / 2, HEIGHT / 4);
+  let orientation = 0;
+  /**
+   * @type {[Point, number][]}
+   */
+  const stack = [];
+  let depth = 0;
+
+  const primitives = [];
+
+  for (const c of tree_str) {
+    if (c === "F") {
+      const forward_angle = Math.PI / 2 + orientation * DELTA_ANGLE;
+      const forward = Point.dir_from_angle(forward_angle);
+      const length = MAX_LENGTH * LENGTH_SCALE ** depth;
+      const start = position;
+      const end = position.add(forward.scale(length));
+      primitives.push(new LinePrimitive(start, end));
+      position = end;
+    } else if (c === "[") {
+      depth++;
+      stack.push([position, orientation]);
+    } else if (c === "]") {
+      depth--;
+      [position, orientation] = stack.pop();
+    } else if (c === "+") {
+      orientation++;
```

Comment:
The animation should be tweening between the previous and next orientation as a float for this section